training/src/app/joke-generator/joke-generator.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
+
+@Component({
+	selector: 'app-joke-generator',
+	templateUrl: './joke-generator.component.html',
+	styleUrls: ['./joke-generator.component.scss']
+})
+export class JokeGeneratorComponent implements OnChanges {
+	@Input() joke: string = ';'
+	@Output() generateJokeClicked : EventEmitter<boolean> = new EventEmitter<boolean>();
+	constructor() {
+	}
+
+	public ngOnChanges(changes: SimpleChanges) {
+	}
+
+	onClick(click : boolean) {
+		this.generateJokeClicked.emit(click);
+	}
+
+	onNotify(joke : string) {
```

Comment:
"onNotify" is not specific enough, give it a more understandabe name.
also add return type